third_party/move/tools/move-smith/src/codegen.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{ast::*, names::Identifier, types::Type};
+use std::vec;
+
+pub trait CodeGenerator {
```

Comment:
I wonder if we could just use the `std::fmt::Display` trait here, which automatically implements the `ToString` trait?

Reply:
The main reason I created this trait is to have the `emit_code_lines` that returns `Vec<String>`. This allows me to easily add spaces for indentation before each line. If only a `String` is returned, then I'll need to split, pad, and concat. I don't know if `std::fmt::Display` would also give this flexibility.

Reply:
Gotcha. `Display` would not give you that flexibility. Let's keep it as is then.